library/gcm.c

```diff
@@ -442,6 +464,11 @@ int mbedtls_gcm_update( mbedtls_gcm_context *ctx,
         return( MBEDTLS_ERR_GCM_BAD_INPUT );
     }
 
+    if ( ( ctx->len == 0 ) && ( ctx->add_len % 16 ) )
```

Comment:
Style:
```suggestion
    if( ctx->len == 0 && ctx->add_len % 16 != 0 )
```
(No space before `(` per style guide. No implicit comparison with 0 per style guide. Not putting redundant parentheses here is my personal preference, I think they mislead the reader into thinking that something unusual is going on, but feel free to leave them if you prefer.)